### ISSUE TITLE
Add Sha256 Checksum to release files

### ISSUE THREAD

**Issue by emoscardini** [openziti/ziti#769]
Wondering if it would be possible to add a checksum for the release binaries.

For example, the binary release: [ziti-darwin-amd64-0.25.13.tar.gz](https://github.com/openziti/ziti/releases/download/v0.25.13/ziti-darwin-amd64-0.25.13.tar.gz)

Could the pipeline create a ziti-darwin-amd64-0.25.13.tar.gz.sha256 file containing the checksum?

**Comment by qrkourier** [comment]
Yes, and this could be a sidecar artifact too (as opposed to checking it in to the repo) with exactly the same name plus the sha256sum suffix `.sha256`

**Comment by qrkourier** [comment]
This ought to do it if you can work with all the checksums in a single file like checksum.txt: https://github.com/marketplace/actions/generate-checksum

**Comment by qrkourier** [comment]
Publish sha256sum-compatible hashes as release artifacts for these repos:

- ziti
- ziti-tunnel-sdk-c
- ziti-sdk-c
